### PR TITLE
fix: correct capabilities:presentation and deviceprofiles:view commands

### PR DESCRIPTION
**Pull request by Sitlintac**
- fix `capabilities:presentation` and `deviceprofiles:view` commands to prompt for user selection if no ID provided
- minor update to utility library for capabilities

## Checklist

- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [ ] Any required documentation has been added
- [x] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: packages/cli/src/commands/capabilities/presentation.ts [hunk 1]
```diff
@@ -82,19 +82,17 @@ export default class PresentationsCommand extends APIOrganizationCommand<typeof
...
-			sortKeyName,
-			listTableFieldDefinitions: ['id', 'version', 'status'],
-			buildTableOutput: (data: CapabilityPresentation) => buildTableOutput(this.tableGenerator, data),
+
+		const capabilityId = await chooseCapability(this, this.args.id, this.args.version, undefined, this.flags.namespace)
+		try {
+			const presentation = await this.client.capabilities.getPresentation(capabilityId.id, capabilityId.version)
+			await formatAndWriteItem(this, { buildTableOutput: data => buildTableOutput(this.tableGenerator, data) }, presentation)
+		} catch (error) {
+			if ('message' in error && error.message.includes('status code 403')) {
```
Comment (rossiam) [inline packages/cli/src/commands/capabilities/presentation.ts:91]: I usually avoid special error handling of errors from the API. The API returns error messages in a standard format (most of the time). We have an issue open to "display standard error messages in a nicer format" which would do something akin to this, but generically for all commands and errors.

Also, if you do want to check for a status code, you should be able to find it more specifically at `error.response.status` like this:

```if (error.response?.status === 403)```

File: packages/cli/src/commands/deviceprofiles/view.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
-import { DeviceProfile } from '@smartthings/core-sdk'
+import { APIOrganizationCommand, formatAndWriteItem, outputItemOrList } from '@smartthings/cli-lib'
```
Comment (rossiam) [inline packages/cli/src/commands/deviceprofiles/view.ts:1]: Similar to `presentation.ts`, we shouldn't need `outputItemOrList` here anymore and the flags should come from `formatAndWriteItem`.

**Review by rossiam — APPROVED** [review]

File: .changeset/violet-moles-think.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@smartthings/cli": patch
+---
+
+fix `capabilities:presentation` and `deviceprofiles:view` commands to wait for user selection
```
[the remark this answers is not in the document]
Comment (Sitlintac) [reply .changeset/violet-moles-think.md:5]: That's much better, thanks!

File: packages/cli/src/commands/capabilities/presentation.ts [hunk 4]
```diff
@@ -2,9 +2,9 @@ import { Flags } from '@oclif/core'
 
 import { CapabilityPresentation } from '@smartthings/core-sdk'
 
-import { APIOrganizationCommand, OutputItemOrListConfig, outputItemOrListGeneric, TableGenerator } from '@smartthings/cli-lib'
+import { APIOrganizationCommand, formatAndWriteItem, outputItemOrListGeneric, TableGenerator } from '@smartthings/cli-lib'
```
[the remark this answers is not in the document]
Comment (Sitlintac) [reply packages/cli/src/commands/capabilities/presentation.ts:5]: It didn't even occur to me to switch out the flags when I changed the output method.  I'll remember to watch out for this in future, too.

File: packages/cli/src/commands/capabilities/presentation.ts (hunk 1 above)
Comment (Sitlintac) [reply packages/cli/src/commands/capabilities/presentation.ts:91]: I've taken out the special handling.  I really prefer having a nicer format to the message, but I see your point that it's better handled uniformly via that other open issue.

File: packages/cli/src/commands/deviceprofiles/view.ts (hunk 2 above)
Comment (Sitlintac) [reply packages/cli/src/commands/deviceprofiles/view.ts:1]: Done.

**Pull request merged**